File: MADVCamera/CameraManager/ambaresponse/AMBAGetMediaInfoResponse.java
package com.madv360.madv.connection.ambaresponse;

import com.google.gson.annotations.Expose;
import com.madv360.madv.connection.AMBAResponse;

/**
 * Created by qiudong on 16/7/7.
 */
public class AMBAGetMediaInfoResponse extends AMBAResponse {
    @Expose
    public int duration;

    @Expose
    private long size;

    public long getSize() {
        if (size < 0)
            return (1L << 32) + size;
        else
            return size;
    }
}

File: MADVCamera/CameraManager/ambaresponse/AMBAGetMediaInfoResponse.m
//
//  AMBAGetMediaInfoResponse.m
//  Madv360_v1
//
//  Created by QiuDong on 16/10/11.
//  Copyright © 2016年 Cyllenge. All rights reserved.
//

#import "AMBAGetMediaInfoResponse.h"

@implementation AMBAGetMediaInfoResponse

+ (NSArray<NSString* >*) jsonSerializablePropertyNames {
    mergeJsonSerializablePropertyNames(array, @[@"duration", @"jsonSize", @"media_type", @"scene_type", @"gyro"]);
    return array;
}

+ (NSDictionary<NSString*, NSString* >*) propertyNameToJsonKeyMap {
    mergePropertyNameToJsonKeyMap(dict, @{@"jsonSize":@"size"});
    return dict;
}

- (NSInteger) size {
    if (self.jsonSize < 0)
        return (1L << 32) + self.jsonSize;
    else
        return self.jsonSize;
}

@end
